spec/memoize_query_spec.rb

```diff
@@ -258,6 +258,30 @@ def expect_to_eq_with_or_without_redis
         end
       end
 
+      if Site.respond_to?(:upsert_all)
```

Comment:
Can we also test when upsert_all is given a conflict target, otherwise it looks like this test just falls back to the insert_all logic.

Reply:
> upsert_all is given a conflict target

Rails does not support that 😄 

> otherwise it looks like this test just falls back to the insert_all logic.

Yeah it's pretty much a "insert all, on duplicate do nothing"

Reply:
Yea we could! I added a comment on top... to remind ourselves to support conflict_target as soon as Rails does.